pandas/_libs/reduction.pyx

```diff
@@ -1,9 +1,24 @@
 # -*- coding: utf-8 -*-
 # cython: profile=False
-import numpy as np
-
 from distutils.version import LooseVersion
 
+from cython cimport Py_ssize_t
+from cpython cimport Py_INCREF
+
+from libc.stdlib cimport malloc, free
+
+import numpy as np
+cimport numpy as np
+from numpy cimport (ndarray,
+                    int64_t,
+                    PyArray_SETITEM,
+                    PyArray_ITER_NEXT, PyArray_ITER_DATA, PyArray_IterNew,
+                    flatiter)
+np.import_array()
+
+cimport util
+from lib import maybe_convert_objects
```

Comment:
remove this, and instead where ``.reduce()`` is called you can do this on the python side.

Reply:
We'd also need to do that everywhere SeriesGrouper.get_result or SeriesBinGrouper.get_result is called.  What's the upside that makes the duplicated code worthwhile?

Reply:
this completely removes this module from the import time dependency chain.

you *could* do this in the top-level function ``reduce()`` I suppose (inside the function)

Reply:
or could expose ``maybe_convert_objects`` in a pxd and then call it directly in cython (but that actually puts this back in the dep chain).

Reply:
> you could do this in the top-level function reduce() I suppose (inside the function)

That would be better than trying to do it after every place in the code-base where the function is called.  But note that `reduce` is not the only part of this module that is used externally, so we'd need to do this run-time import in three places in this module.

Since the dependency is one-way and not a c-dep, I don't see a huge downside to the import-time import of maybe_convert_objects.